embassy-sync/src/waitqueue/atomic_waker.rs

```diff
@@ -30,12 +30,19 @@ impl<M: RawMutex> GenericAtomicWaker<M> {
         })
     }
 
+    /// Removes previously registered waker.
+    pub fn deregister(&self) {
```

Comment:
This could be
```suggestion
    pub fn take(&self) -> Option<Waker> {
```
which pretty much does the same thing but is much more useful.